src/test/groovy/com/twcable/grabbit/jcr/AuthorizableProtoNodeDecoratorSpec.groovy

```diff
@@ -0,0 +1,339 @@
+                .build()
+        nodeBuilder.addProperties(authorizableIdProperty)
+
+        ProtoProperty authorizableCategory = ProtoProperty
+                .newBuilder()
+                .setName('cq:authorizableCategory')
+                .setType(STRING)
+                .setMultiple(false)
+                .addValues(ProtoValue.newBuilder().setStringValue('mcm'))
+                .build()
+        nodeBuilder.addProperties(authorizableCategory)
+
+        ProtoProperty simplePrimaryType = ProtoProperty
+                .newBuilder()
+                .setName(JCR_PRIMARYTYPE)
+                .setType(STRING)
+                .setMultiple(false)
+                .addValues(ProtoValue.newBuilder().setStringValue('nt:unstructured'))
+                .build()
+        nodeBuilder.addProperties(authorizableCategory)
+        if(hasPreferences) {
+            ProtoNode preferenceNode = ProtoNode.newBuilder()
+                .setName('preferences')
+                .addProperties(simplePrimaryType)
+                .build()
+            nodeBuilder.addMandatoryChildNode(preferenceNode)
+        }
+        if(hasProfile) {
+            ProtoNode profileNode = ProtoNode
+                .newBuilder()
+                .setName('profile')
+                .addProperties(simplePrimaryType)
+                .build()
+            nodeBuilder.addMandatoryChildNode(profileNode)
+        }
+        final properties = [new ProtoPropertyDecorator(primaryTypeProperty), new ProtoPropertyDecorator(disabledProperty), new ProtoPropertyDecorator(authorizableIdProperty), new ProtoPropertyDecorator(authorizableCategory)]
+        return GroovySpy(AuthorizableProtoNodeDecorator, constructorArgs: [nodeBuilder.build(), properties], configuration)
+    }
+
+
+    def "Throws an InsufficientGrabbitPrivilegeException if JVM permissions are not present"() {
+        when:
+        final protoNodeDecorator = theProtoNodeDecorator(false, false, false) {
+            it.getSecurityManager() >> Mock(SecurityManager) {
+                it.checkPermission(permission) >> {
+                    throw new SecurityException()
+                }
+            }
+        }
+
+        protoNodeDecorator.writeToJcr(Mock(Session))
+
+        then:
+        thrown(InsufficientGrabbitPrivilegeException)
+
+        where:
+        permission << [new ReflectPermission('suppressAccessChecks'), new RuntimePermission('accessDeclaredMembers'), new RuntimePermission('accessClassInPackage.{org.apache.jackrabbit.oak.security.user}')]
+    }
+
+
+    def "Passes security check if all JVM permissions are present"() {
+        when:
+        final session = Mock(Session) {
+            it.getNode('authorizablePath') >> Mock(Node) {
+                getProperty(JCR_PRIMARYTYPE) >> Mock(Property) {
+                    it.getString() >> 'nt:unstructured'
+                }
+                getProperties() >> Mock(PropertyIterator) {
+                    it.toList() >> []
```

Comment:
yeah, it would return 'null', but toList() never returns null. So it would artificially break @ JcrNodeDecorator:54